app/src/main/java/io/left/meshim/controllers/RightMeshController.java

```diff
@@ -236,11 +242,12 @@ private void handlePeerChanged(RightMeshEvent e) {
         PeerChangedEvent event = (PeerChangedEvent) e;
 
         // Ignore ourselves.
-        if (event.peerUuid == user.getMeshId()) {
+        if (event.peerUuid.equals(meshManager.getUuid())) {
             return;
         }
 
-        if (event.state == ADDED) {
+        if (event.state != REMOVED && !discovered.contains(event.peerUuid)) {
```

Comment:
Unless `UPDATED`  isn't a thing anymore, this will now send an update payload for everytime someone's network state changes as well.

Also since this is public and the behaviour of this line isn't clear without knowing what the other events are this should probably be commented.

Reply:
D'oh, just read your description. Are we hoping to fix that in the library?

I'd still recommend `event.state == ADDED || event.state == UPDATED` though. More clear/readable.

Reply:
Yeah ideally it will get fixed, but its a really complicated buy so in order to make a good demo this is second best fix. Good idea though, it will be clearer what is happening if we do this.